corehq/apps/email/forms.py

```diff
@@ -0,0 +1,164 @@
+                InlineField('action'),
+                Div(InlineField('hq_api_id', css_class="ko-select2"), css_class='col-sm-6 col-md-6 col-lg-4'),
+                Div(StrictButton(
+                    mark_safe('<i class="fa fa-plus"></i> Add Another Gateway'),  # nosec: no user input
+                    css_class='btn-primary',
+                    type='submit',
+                    style="margin-left:5px;"
+                ), css_class='col-sm-3 col-md-2 col-lg-2'),
+            ),
+        )
+
+    @staticmethod
+    def email_backend_classes():
+        backends = copy.deepcopy(get_email_backend_classes())
+        return backends
+
+
+class BackendForm(Form):
+    domain = None
+    backend_id = None
+    name = CharField(
+        label=gettext_noop("Name"),
+        required=True,
+    )
+    display_name = CharField(
+        label=gettext_noop("Display Name"),
+        required=False,
+    )
+    description = CharField(
+        label=gettext_noop("Description"),
+        widget=forms.Textarea(attrs={"class": "vertical-resize"}),
+        required=False,
+    )
+    username = CharField(
+        label=gettext_noop("Username"),
+        required=True,
+    )
+    password = CharField(
+        label=gettext_noop("Password"),
+        required=True,
+        widget=forms.PasswordInput(),
+    )
+    server = CharField(
+        label=gettext_noop("Server"),
+        required=True,
+    )
+    port = CharField(
+        label=gettext_noop("Port"),
+        required=True,
+    )
+
+    @property
+    def general_fields(self):
+        fields = [
+            crispy.Field('name', css_class='input-xxlarge'),
+            crispy.Field('display_name', css_class='input-xxlarge'),
+            crispy.Field('description', css_class='input-xlarge'),
+            crispy.Field('username', css_class='input-xxlarge'),
+            crispy.Field('password', css_class='input-xxlarge'),
+            crispy.Field('server', css_class='input-xxlarge'),
+            crispy.Field('port', css_class='input-xxlarge'),
+        ]
+
+        return fields
+
+    def __init__(self, *args, **kwargs):
+        button_text = kwargs.pop('button_text', _("Create Email Gateway"))
+        self.domain = kwargs.pop('domain')
+        self.backend_id = kwargs.pop('backend_id')
+        super(BackendForm, self).__init__(*args, **kwargs)
+        self.helper = HQFormHelper()
+        self.helper.form_method = 'POST'
+        self.helper.layout = crispy.Layout(
+            crispy.Fieldset(
+                _('General Settings'),
+                *self.general_fields
+            ),
+            self.gateway_specific_fields,
+            hqcrispy.FormActions(
+                StrictButton(
+                    button_text,
+                    type="submit",
+                    css_class='btn-primary'
+                ),
+            ),
+        )
+
+        if self.backend_id:
+            #   When editing, don't allow changing the name because name might be
```

Comment:
Shouldn't we throw an error in the `clean_name` when name is being modified for an existing backend?